libsolidity/parsing/Parser.cpp

```diff
@@ -370,6 +370,24 @@ std::pair<ContractKind, bool> Parser::parseContractKind()
 	return std::make_pair(kind, abstract);
 }
 
+ASTPointer<Expression> Parser::parseContractStorageBaseLocationExpression()
+{
+	solAssert(m_scanner->currentLiteral() == "layout");
+	expectToken(Token::Identifier);
+	if (
+		m_scanner->currentToken() != Token::Identifier ||
```

Comment:
> Identifier check is redundant here - it'll throw a fatal parsing error in expectToken(...) if it's not an identifier.

 commented previously [here](https://github.com/ethereum/solidity/pull/15528#discussion_r1808363997).

@nikola-matic , I am not sure if you were suggesting that we use `expectToken` instead of `if` to check the identifier with literal `at`.
I used `if` here because we also need to check for the literal value `at`.
Or were you referring to `expectToken` in line `376`? (in which case, it actually is consuming the identifier with literal `layout`).

Reply:
Yup, you are correct. You could do the same as above (check that literal is `at` and then `expectToken(Identifier)`, but then you'd just get the generic error `Expected identifier but got...`, and since we're not introducing new tokens (at, layout), it makes sense for error to be specifically tailored like it is here.

One suggestion I could make is to have the error follow the general `expectToken()` convention, I.e. `Expected 'at' but got '..'`, since I assume we're gonna introduce layout and at tokens (as well as transient) for the next breaking release.

Reply:
> One suggestion I could make is to have the error follow the general `expectToken()` convention

Yeah, I will make this change.